File: backend/src/main/java/com/dbv/scoutmission/entity/Submission.java
package com.dbv.scoutmission.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "submissions")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class Submission {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id", nullable = false)
    private Team team;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "challenge_id", nullable = false)
    private Challenge challenge;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "base_id", nullable = false)
    private Base base;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String answer;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, columnDefinition = "submission_status")
    private SubmissionStatus status;

    @Column(name = "submitted_at", nullable = false)
    private Instant submittedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reviewed_by")
    private User reviewedBy;

    @Column(columnDefinition = "TEXT")
    private String feedback;

    @PrePersist
    protected void onCreate() {
        if (submittedAt == null) submittedAt = Instant.now();
        if (status == null) status = SubmissionStatus.pending;
    }
}

File: web/src/test/factories/submission.ts
import type { Submission } from '@/types'

let counter = 0

export function createMockSubmission(overrides?: Partial<Submission>): Submission {
  counter++
  return {
    id: `sub-${counter}`,
    teamId: 'team-1',
    challengeId: 'challenge-1',
    baseId: 'base-1',
    answer: `Answer ${counter}`,
    status: 'pending',
    submittedAt: new Date().toISOString(),
    ...overrides,
  }
}

export function resetSubmissionCounter() {
  counter = 0
}
